06_puppy/kasunfdo/mapstore.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"fmt"
+)
+
+// MapStore is map based implementation of Storer interface
+type MapStore struct {
+	store  map[uint64]Puppy
+	nextID uint64
+}
+
+// NewMapStore creates a new MapStore
+func NewMapStore() *MapStore {
+	return &MapStore{nextID: 0, store: map[uint64]Puppy{}}
+}
+
+// CreatePuppy inserts given puppy in store and returns given id to the puppy
+func (m *MapStore) CreatePuppy(puppy Puppy) uint64 {
+	m.nextID++
+	puppy.ID = m.nextID
+	m.store[puppy.ID] = puppy
+	return puppy.ID
+}
+
+// ReadPuppy reads puppy with given id from the store
+func (m *MapStore) ReadPuppy(id uint64) (Puppy, error) {
+	if puppy, exists := m.store[id]; exists {
+		return puppy, nil
```

Comment:
Very nit picky, but imo logic should be inverted: if !exist ...
Is bail early on error. below too

Reply:
I did in this way because otherwise I'll have to use `else` to retain `puppy` value or I'll have to make `puppy, exists := m.store[id]` a statement.

I'll change as requested.